plutus-core/budgeting-bench/Bench.hs

```diff
@@ -50,21 +56,21 @@ createTwoTermBuiltinBench name as bs =
     bgroup (show name) $
         as <&> (\(x, xMem) ->
             bgroup (show xMem) $ bs <&> (\(y, yMem) ->
-                runTermBench (show yMem) $ mkIterApp () (staticBuiltinNameAsTerm name) [(mkConstant () x), (mkConstant () y)]
+                runTermBench (show yMem) $ erase $ mkIterApp () (staticBuiltinNameAsTerm name) [(mkConstant () x), (mkConstant () y)]
```

Comment:
I suppose we should really have untyped versions of `mkIterApp` and so on and construct untyped terms directly rather than constructing typed terms and then erasing them. (Or should we? Maybe the typed ones are more likely to be correct?). Any opinions, @effectfully ?

Anyway, that's definitely not something for this PR: what's here does what's needed with minimal messing about.

Reply:
Either way is fine, but not having the additional indirection of going through typed terms would be a bit better in my view.